Why would we get 'Cannot invoke "java.io.File.toURI()" because "file" is null' from Maven plugin?

We've been using "jaxb-maven-plugin" for a long time to generate JAXB classes from wsdls and xsds. There are occasional unusual issues we have to deal with, but we've always gotten it to work. We maintain a large number of SpringBoot services, many of which use this to generate JAXB classes from many different schemas.

Today I started looking at one build that is failing with an odd error:

    14:26:52 [ERROR] Failed to execute goal org.jvnet.jaxb:jaxb-maven-plugin:4.0.8:generate (ivsad) on project UnifiedAccountMs: Execution ivsad of goal org.jvnet.jaxb:jaxb-maven-plugin:4.0.8:generate failed: Cannot invoke "java.io.File.toURI()" because "file" is null -> [Help 1]

This pom.xml has several "execution" blocks, but I've tried commenting all but one, and other variations, but all of those experiments fail with the same error (with a different execution name).

I've verified that the "sourceDirectory" refers to an existing directory that has a wsdl in it, with several xsd files. Each execution block looks something like this:
```  
  <execution>
	<id>unifiedServices</id>
	<goals>
	  <goal>generate</goal>
	</goals>
	<configuration>
	  <specVersion>2.2</specVersion>
	  <schemaDirectory>${project.basedir}/src/main/resources/schemas/csi_UnifiedServices_240.0_schema</schemaDirectory>
	  <schemaIncludes>
		<include>*.wsdl</include>
	  </schemaIncludes>
	  <generateDirectory>${project.build.directory}/generated-sources/xjc/unifiedServices/</generateDirectory>
	  <forceRegenerate>false</forceRegenerate>
	</configuration>
  </execution>
```

What are possible causes of this?

--- a/maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/RawXJCMojo.java
+++ b/maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/RawXJCMojo.java
@@ -397,7 +397,7 @@ public abstract class RawXJCMojo<O> extends AbstractXJCMojo<O> {
 		this.xjcPluginArtifacts = ArtifactUtils.resolveTransitively(
             getArtifactFactory(), getRepositorySystem(),
             getMavenSession().getLocalRepository(), getArtifactMetadataSource(),
-            getPlugins(), getProject(), getArtifactExcludes());
+            getPlugins(), getProject(), getLog(), getArtifactExcludes());
 		this.xjcPluginFiles = ArtifactUtils.getFiles(this.xjcPluginArtifacts);
 		this.xjcPluginURLs = CollectionUtils.apply(this.xjcPluginFiles, IOUtils.GET_URL);
 	}

--- a/maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/util/ArtifactUtils.java
+++ b/maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/util/ArtifactUtils.java
@@ -17,6 +17,8 @@ import org.apache.maven.artifact.resolver.ArtifactResolutionRequest;
 import org.apache.maven.artifact.resolver.ArtifactResolutionResult;
 import org.apache.maven.artifact.resolver.filter.ExclusionSetFilter;
 import org.apache.maven.model.Dependency;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.plugin.logging.SystemStreamLog;
 import org.apache.maven.project.MavenProject;
 import org.apache.maven.project.artifact.InvalidDependencyVersionException;
 import org.apache.maven.project.artifact.MavenMetadataSource;
@@ -40,6 +42,7 @@ public class ArtifactUtils {
             artifactFactory, artifactResolver,
             localRepository, artifactMetadataSource,
             dependencies, project,
+            new SystemStreamLog(),
             null);
     }
 
@@ -49,10 +52,11 @@ public class ArtifactUtils {
 			final ArtifactRepository localRepository,
 			final ArtifactMetadataSource artifactMetadataSource,
 			final Dependency[] dependencies, final MavenProject project,
+            final Log log,
             final String[] artifactExcludes)
 			throws InvalidDependencyVersionException,
 			ArtifactResolutionException, ArtifactNotFoundException {
-		if (dependencies == null) {
+		if (dependencies == null || dependencies.length == 0) {
 			return Collections.emptyList();
 		}
 
@@ -74,9 +78,19 @@ public class ArtifactUtils {
 
         final ArtifactResolutionResult artifactResolutionResult = artifactResolver.resolve(request);
 
-		final Set<Artifact> resolvedArtifacts = artifactResolutionResult.getArtifacts();
-
-		return resolvedArtifacts;
+        if (!artifactResolutionResult.isSuccess()) {
+            if (artifactResolutionResult.hasMissingArtifacts()) {
+                log.error("artifactResolutionResult has missing artifacts : " + artifactResolutionResult.getMissingArtifacts());
+                throw new RuntimeException("Artifacts - missing artifacts");
+            } else if (artifactResolutionResult.hasExceptions()) {
+                log.error("artifactResolutionResult has exceptions : " + artifactResolutionResult.getExceptions());
+                throw new RuntimeException("Artifacts - exceptions", artifactResolutionResult.getExceptions().get(0));
+            } else {
+                log.error("artifactResolutionResult status is not success");
+                throw new RuntimeException("Artifacts - not in success");
+            }
+        }
+        return artifactResolutionResult.getArtifacts();
 	}
 
 	public static Collection<Artifact> resolve(
@@ -87,7 +101,7 @@ public class ArtifactUtils {
 			final Dependency[] dependencies, final MavenProject project)
 			throws InvalidDependencyVersionException,
 			ArtifactResolutionException, ArtifactNotFoundException {
-		if (dependencies == null) {
+		if (dependencies == null || dependencies.length == 0) {
 			return Collections.emptyList();
 		}
 

